### PR TITLE
Added Dremio source to Data Integration Pipelines

### PR DESCRIPTION
**Pull request by Luishfs**
# Description

Added Dremio Source to Data Integrations Pipelines

# How Has This Been Tested?

This Source was tested using a Local Dremio instance -> PostgreSQL destination
The data source used in Dremio was a PostgreSQL Table, so i could compare if data/schema writing was matching PostgreSQL Source.

## Schema Comparison
Default Table Schema:
Screenshot from 2023-11-25 03-51-54

Dremio Source Table Schema:
Screenshot from 2023-11-25 03-52-12

## Test Connection
Success:
Screenshot from 2023-11-25 03-44-17

Failed password/username:
Screenshot from 2023-11-25 03-44-31

Failed host:
Screenshot from 2023-11-25 03-44-48

## FULL TABLE sync
Screenshot from 2023-11-25 03-53-02

## INCREMENTAL sync
Screenshot from 2023-11-25 03-53-18

Screenshot from 2023-11-25 03-53-30

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

cc:

@wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/sources/dremio/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+| `username` | Username of the Dremio account to use for authenticating. | True |
+| `schema` | Dremio Source Table schema. | True |
+| `source_backend` | SQL Source backend used to query data from Dremio. Please refer to the section below for detailed info <br /> E.G postgresql, mssql, mysql. | True, if applicable |
+| `password` | Password of the Dremio account to use for authenticating. | True, if no token is provided. |
+| `hostname` | The hostname or IP address of the coordinator node. | False (Default is localhost) |
+| `port` | Dremio's Arrow Flight server port. Can be other than 32010, if changed on the coordinator node. | False (Default is 32010) |
+| `token` | Either a Personal Access Token or an OAuth2 Token. | True, if no password is provided. |
+| `tls` | Enables encryption on a connection. | False |
+| `disable_certificate_verification` | Disables TLS server verification. | False |
+| `path_to_certs` | Path to trusted certificates for encrypted connections. False |
```
Comment (shrutimantri) [inline mage_integrations/mage_integrations/sources/dremio/README.md:25]: A pipe is missing before False in this line.

Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/dremio/README.md]: Fixed in c78c0c2014babbe93ae7c3f20d2a522aeec2ff48
Thanks for the heads up!

File: mage_integrations/mage_integrations/sources/dremio/templates/config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "username": "",
+    "password": "",
+    "hostname": "localhost",
+    "port": 32010
+}
```
[the remark this answers is not in the document]
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/dremio/templates/config.json]: Fixed in c78c0c2014babbe93ae7c3f20d2a522aeec2ff48

File: mage_integrations/mage_integrations/sources/dremio/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+This source uses Dremio Arrow Flight Client to connect/execute SQL queries and shares most of its settings.
+To find more details, please refer to Dremio docs [here](https://docs.dremio.com/current/sonar/client-applications/clients/python)
+
+<b>NOTE:</b> Mage's implementation of Dremio Source doesn't allow for the `query` connection parameter
+
+| Key | Description | Required |
+| --- | --- | --- |
+| `username` | Username of the Dremio account to use for authenticating. | True |
+| `schema` | Dremio Source Table schema. | True |
+| `source_backend` | SQL Source backend used to query data from Dremio. Please refer to the section below for detailed info <br /> E.G postgresql, mssql, mysql. | True, if applicable |
```
[the remark this answers is not in the document]
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/dremio/README.md]: Modified in c78c0c2014babbe93ae7c3f20d2a522aeec2ff48

File: Dockerfile [hunk 4]
```diff
@@ -35,6 +35,7 @@ RUN \
 # Mage integrations and other related packages
 RUN \
   pip3 install --no-cache-dir "git+https://github.com/wbond/oscrypto.git@d5f3437ed24257895ae1edd9e503cfb352e635a8" && \
+  pip3 install --no-cache-dir "git+https://github.com/dremio-hub/arrow-flight-client-examples.git#egg=dremio-flight&subdirectory=python/dremio-flight" && \
```
[the remark this answers is not in the document]
Comment (Luishfs) [reply Dockerfile:38]: Not that i'm aware of.
The oficial docs redirect to this [repo](https://github.com/dremio-hub/arrow-flight-client-examples/tree/main/python) where it contains a release with a .whl file, but this file is outdaded, hence why i pip installed it using the oficial repo path

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**